python_modules/dagster/dagster/_core/definitions/mutable_partitions_definition.py

```diff
@@ -0,0 +1,86 @@
+from datetime import datetime
+from typing import Optional, Sequence
+
+import dagster._check as check
+from dagster._annotations import experimental
+from dagster._core.instance import DagsterInstance
+
+from .partition import (
+    Partition,
+    PartitionsDefinition,
+    raise_error_on_invalid_partition_key_substring,
+)
+
+
+@experimental
+class MutablePartitionsDefinition(PartitionsDefinition):
+    """
+    A partitions definition whose partition keys can be dynamically added and removed.
+
+    This is useful for cases where the set of partitions is not known at definition time,
+    but is instead determined at runtime.
+
+    Partitions can be added and removed using the `add_partitions` and `remove_partitions` methods.
+    For example:
+
+    ```
```

Comment:
We should use the standard sphinx examples syntax.  Look at `@asset` for an example.